### PR TITLE
PopPUNK v2.5.0

### PR DESCRIPTION
**Pull request by stitam**
Hi,

PopPUNK v2.4.0 was already on StaPH-B but a new release was issued on 25 Aug 2022. This PR uses the Dockerfile and README.md of PopPUNK v2.4.0 as templates and updates them to v2.5.0.

Changes after line 48 of the Dockerfile were necessary because the tools `poppunk_prune` and `poppunk_tsne` seem to be missing from v.2.5.0 and this prevented the image from building successfully. The updated image builds successfully on my platform.

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [ ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed. (Maybe not relevant because this is an update to a previously accepted version?)
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by stitam** [comment]
While all included tests pass, when I run `poppunk --create-db` or `poppunk --fit-model` I get the following error:

```
Traceback (most recent call last):
  File "/opt/conda/envs/poppunk-env/bin/poppunk", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/envs/poppunk-env/lib/python3.10/site-packages/PopPUNK/__main__.py", line 213, in main
    from .models import loadClusterFit, BGMMFit, DBSCANFit, RefineFit, LineageFit
  File "/opt/conda/envs/poppunk-env/lib/python3.10/site-packages/PopPUNK/models.py", line 19, in <module>
    import hdbscan
  File "/opt/conda/envs/poppunk-env/lib/python3.10/site-packages/hdbscan/__init__.py", line 1, in <module>
    from .hdbscan_ import HDBSCAN, hdbscan
  File "/opt/conda/envs/poppunk-env/lib/python3.10/site-packages/hdbscan/hdbscan_.py", line 509, in <module>
    memory=Memory(cachedir=None, verbose=0),
TypeError: Memory.__init__() got an unexpected keyword argument 'cachedir'
```
The assemblies which produce this error  can be processed without any errors with both `staphb/poppunk:2.4.0` and the developers' own `mrcide/poppunk:bacpop-20` containers (which is also 2.4.0). I am wondering whether the issue is with the container instructions or the tool itself: I am not a python expert but it seem like the issue is with the code. What do you think?

**Comment by stitam** [comment]
Thanks @kapsakcj! Also this issue with hsbscan might be related: https://github.com/scikit-learn-contrib/hdbscan/issues/565

**Comment by stitam** [comment]
In the end I added `joblib==1.1.0` to the micromamba call and now the container seems to be working. Please note that this is only a temporary solution because joblib v1.1.0 has a known vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2022-21797

File: poppunk/2.5.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+LABEL base.image="mambaorg/micromamba:0.24.0"
+LABEL dockerfile.version="1"
+LABEL software="PopPUNK"
+LABEL software.version=${POPPUNK_VERSION}
+LABEL description="POPulation Partitioning Using Nucleotide Kmers"
+LABEL website="https://github.com/bacpop/PopPUNK"
+LABEL license="https://github.com/bacpop/PopPUNK/blob/master/LICENSE"
+LABEL maintainer1="Curtis Kapsak"
+LABEL maintainer1.email="curtis.kapsak@theiagen.com"
```
Comment (kapsakcj) [inline poppunk/2.5.0/Dockerfile:20]: Would you like to add yourself here as a maintainer? `LABEL maintainer2="..."`

**Comment by kapsakcj** [comment]
Gotcha, thanks for the heads up about joblib and the vulnerability. I'm OK with including it, as it would be quite a challenge to incorporate a newer, non-vulnerable version. That should be handled by the poppunk dev team.

I tested this docker image with a Strep pneumo genome and the GPS database and everything ran as expected:
```bash
$ poppunk_assign --db GPS_v6 --distances GPS_v6/GPS_v6.dists --query /data/SRR2912551_poppunk_input.tsv --output docker_2.5.0_test --external-clustering GPS_v6_external_clusters.csv
PopPUNK: assign
        (with backend: sketchlib v2.0.0
         sketchlib: /opt/conda/envs/poppunk-env/lib/python3.10/site-packages/pp_sketchlib.cpython-310-x86_64-linux-gnu.so)
Mode: Assigning clusters of query sequences


Graph-tools OpenMP parallelisation enabled: with 1 threads
Sketching 1 genomes using 1 thread(s)
Progress (CPU): 1 / 1
Writing sketches to file
Loading previously refined model
Completed model loading
WARNING: versions of input databases sketches are different, results may not be compatible
Calculating distances using 1 thread(s)
Progress (CPU): 100.0%
Loading network from GPS_v6/GPS_v6_graph.gt
Network loaded: 42163 samples

Done
```

I'm good to merge & deploy the docker image if it works in your hands too!

**Comment by johnlees** [comment]
Commenting here as requested via email: we have an issue for the HDBSCAN problem (https://github.com/bacpop/PopPUNK/issues/227). Indeed, the only solution currently is to pin the version of joblib. I am hoping the hdbscan authors push a new version as soon as possible (they have actually fixed the incompatibility, just not released yet).

File: poppunk/2.5.0/Dockerfile (hunk 1 above)
Comment (stitam) [reply poppunk/2.5.0/Dockerfile:20]: Thanks @kapsakcj, but I think it is better to list just one maintainer and this is 95% your code. I am okay with not being mentioned on this one :)

**Comment by stitam** [comment]
Great, so hopefully this will be fixed soon and then we can update the docker file/image. Thanks @johnlees for the feedback!

**Review by kapsakcj — APPROVED** [review]

**Comment by kapsakcj** [comment]
Docker image for poppunk 2.5.0 is now available on dockerhub and quay. The `:latest` docker image tag was overwritten with this new version as well. Thanks for the PR 🎉 

https://hub.docker.com/r/staphb/poppunk/tags
https://quay.io/repository/staphb/poppunk?tab=tags

**Pull request merged**